### ISSUE TITLE
Integration AppIntro into Androidx navigation component

### ISSUE THREAD

**Issue by oliexdev** [AppIntro/AppIntro#801]
**AppIntro Version**: 6.0.0

**Device/Android Version**: Pixel 2

**Issue details / Repro steps / Use case background**: 

For the new [Androidx navigation framework](https://developer.android.com/guide/navigation/navigation-getting-started) the activity needs a navHost fragment. Is it possible to add that into AppIntro?

Or could the AppIntro class also inherited from a Fragment class?

The problem that I am facing is that I want to navigate to another Fragment (outside of the AppIntro acitivty) in my main nav graph with safe arguments but I can't do it without having a second graph in AppIntro or disabling the second AppIntro activity.

Does any have an idea how to solve this issue?

**Comment by cortinico** [comment]
Thanks for raising this issue @oliexdev 

Currently there is no direct support for the Navigation framework. We definitely need to look into this and provide better guidance on how to integrate an AppIntro Activity inside a navigation graph.

> Or could the AppIntro class also inherited from a Fragment class?

Not possible at the moment and also really unlikely in the foreseeable future.

> The problem that I am facing is that I want to navigate to another Fragment (outside of the AppIntro acitivty) in my main nav graph with safe arguments but I can't do it without having a second graph in AppIntro or disabling the second AppIntro activity.

Can ask you to be more clear here?

> Does any have an idea how to solve this issue?

Ideally treat your whole AppIntro Activity as a separate entity. Once you started it, you let the user finish the flow. Once the AppIntro is over, you can navigate to the entry point of your nav graph.

**Comment by oliexdev** [comment]
Thank for your quick reply. :trophy: 

> Can ask you to be more clear here?

I have a UserSetting fragment class which is used in my app. During the AppIntro slide I added a custom button which should open the UserSetting fragment. So I can add the user during the app intro slide, see [my app](https://f-droid.org/repository/browse/?fdid=com.health.openscale) for an example.

> Ideally treat your whole AppIntro Activity as a separate entity. Once you started it, you let the user finish the flow. Once the AppIntro is over, you can navigate to the entry point of your nav graph.

That will only work if I don't interact with my other fragments but the goal is to interact with other fragments outside of the AppIntro activity using the new Androidx navigation component.

**Comment by cortinico** [comment]
> but the goal is to interact with other fragments outside of the AppIntro activity using the new Androidx navigation component

Yeah that's exactly the point that is not supported at the moment.

**Comment by WolandPL** [comment]
I implemented a simple workaround for that issue in my project. I created an empty fragment and put it into my nav graph. In that fragment (with no layout) I start the AppIntro activity for result and after AppIntro activity finishes, in onActivityResult of the empty fragment, I navigate wherever I need to.

**Comment by oliexdev** [comment]
@WolandPL 
thanks you gave me the hint how to implement a workaround. I implemented a slightly different version
at https://github.com/oliexdev/openScale/commit/83e55e05bc9d93f4ba4df7a64b3a44b8b70710b1

@cortinico Should I try to create a PR for AppIntro that includes the required NavFragment?

**Comment by cortinico** [comment]
Sure I'd be happy to review it 👍

**Comment by oliexdev** [comment]
@cortinico I created a PR please review it
